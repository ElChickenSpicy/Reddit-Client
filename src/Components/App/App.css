/* http://meyerweb.com/eric/tools/css/reset/ 
   v2.0 | 20110126
   License: none (public domain)
*/

/* || CSS Reset */
html, body, div, span, applet, object, iframe,
h1, h2, h3, h4, h5, h6, p, blockquote, pre,
a, abbr, acronym, address, big, cite, code,
del, dfn, em, img, ins, kbd, q, s, samp,
small, strike, strong, sub, sup, tt, var,
b, u, i, center,
dl, dt, dd, ol, ul, li,
fieldset, form, label, legend,
table, caption, tbody, tfoot, thead, tr, th, td,
article, aside, canvas, details, embed, 
figure, figcaption, footer, header, hgroup, 
menu, nav, output, ruby, section, summary,
time, mark, audio, video {
	margin: 0;
	padding: 0;
	border: 0;
	font-size: 100%;
	font: inherit;
	vertical-align: baseline;
}
/* HTML5 display-role reset for older browsers */
article, aside, details, figcaption, figure, 
footer, header, hgroup, menu, nav, section {
	display: block;
}
body {
	line-height: 1;
}
ol, ul {
	list-style: none;
}
blockquote, q {
	quotes: none;
}
blockquote:before, blockquote:after,
q:before, q:after {
	content: '';
	content: none;
}
table {
	border-collapse: collapse;
	border-spacing: 0;
}

a {
  text-decoration: none;

}

::-moz-selection { /* Code for Firefox */
  background: rgba(255, 153, 0, 0.26);
}

::selection {
  background: rgba(255, 153, 0, 0.26);
}

/* || Home Page Layout - Desktop  */
.top-container {
  height: 99vh;
  overflow: auto;
  display: flex;
  justify-content: flex-end;
}

nav {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  width: 18%;
  height: 95%;
  margin-top: 10px;
  position: fixed;
  left: 1.5%;
  background-color: lightgray;
}

main {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  border: 3px solid turquoise;
  width: 77%;
  height: fit-content;
  min-height: 100%;
  margin-bottom: 10px;
  margin-right: 1.5%;
  background-color: lightgray;
  padding-top: 90px;
}

main #posts {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.searchbar {
  width: 95%;
  height: 30px;
  border: 2px solid black;
  border-radius: 4px;
  transition-duration: 0.4s;
}

.searchbar:hover {
  background-image: linear-gradient(319deg, #bbff99 0%, #ffec99 37%, #ff9999 100%);
}

.subreddit-container {
  border: 2px solid black;
  width: 95%;
  height: 90%;
  background-color: white;
  display: flex;
  flex-direction: column;
  align-items: center;
  border-radius: 4px;
}

.subreddit-title {
  border: 2px solid black;
  width: 95%;
}

ul {
  border: 1px solid red;
  width: 95%;
  height: 95%;
  margin-top: 20px;
}

ul li {
  margin-bottom: 15px;
  cursor: pointer;
  display: flex;
  align-items: center;
}

ul li img {
  margin-right: 7px;
  height: 30px;
  border-radius: 50%;
}

.main-header {
  border: 2px solid black;
  height: 90px;
  width: 76%;
  position: fixed;
  background-color: white;
  top: 0px;
}

.main-header #brand {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 30px;
  width: 30%;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.main-header img {
  border: 1px solid orange;
  margin-right: 5px;
}

.main-header h2 {
  position: absolute;
  top: 50%;
  transform: translate(0, -50%);
  margin-left: 30px;
  font-weight: 700;
  font-size: 20px;
}

.main-header input {
  display: none;
}

.main-header select {
  display: none;
}

.reddit-post {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border: 2px solid black;
  width: 98%;
  height: fit-content;
  margin-top: 15px;
  background-color: white;
  border-radius: 8px;
}

.post-flex-item {
  display: flex;
  width: 98%;
}

.sub {
  height: 40px;
  margin: 7px 0px;
  justify-content: flex-start;
  align-items: center;
}

.sub img {
  margin-right: 7px;
  height: 30px;
  border-radius: 50%;
}

.content {
  height: 100%;
  background-color: lightgray;
}

.content-title {
  width: 40%;
  border: 1px solid black;
  background-color: white;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  padding-left: 30px;
  padding-right: 30px;
}

.content-media {
  width: 60%;
  margin-top: 0px;
  border: 1px solid black;
  background-color: white;
  text-align: center;
}

.content-link {
  width: 100%;
  margin-top: 0px;
  padding: 40px 20px;
  border: 1px solid black;
  background-color: white;
  color: black;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.content-link:link {
  text-decoration: none;
}

.content-link:hover {
  text-decoration: underline;
}

.thumbnail-container {
  width: 33%;
  margin-left: 20px;
}

.thumbnail {
  height: 120px;
  width: auto;
  border-radius: 8px;
}

.image-container {
  width: 60%;
  display: flex;
  justify-content: center;
  background-color: white;
}

.content-image {
  max-width: 90%;
  max-height: 800px;
}

.content-video {
  max-width: 90%;
  max-height: 800px;
}

.content-text {
  background-color: white;
  width: 60%;
  padding: 50px 30px;
}

.options {
  height: 40px;
  margin: 7px 0px;
}

.options button {
  margin-right: 15px;
  height: 100%;
}

.vote {
  width: 28px;
  padding-bottom: 7px;
  background-color: white;
  border-radius: 8px;
  background-size: contain;
  background-repeat: no-repeat;
  background-position: 0%;
  border: none;
  cursor: pointer;
  transition-duration: 0.2s;
  outline: none;
}

.up {
  background-image: url('../../Icons/up-icon.png');
}

.up:hover {
  background-image: url('../../Icons/up-hover.webp');
}

.down {
  background-image: url('../../Icons/down-icon.png');
}

.down:hover {
  background-image: url('../../Icons/down-hover.webp');
}

.comment-button {
  width: 35px;
  padding-bottom: 2px;
  margin-left: 4px;
  background-color: white;
  border-radius: 8px;
  background-size: 100%;
  background-repeat: no-repeat;
  background-position: 0%;
  border: none;
  cursor: pointer;
  transition-duration: 0.2s;
  outline: none;
}

.home {
  background-image: url('../../Icons/noun_icon.png');
}

.pressed {
  background-image: url('../../Icons/noun_hover.webp');
  cursor: default;
}

.comment-button:hover {
  background-image: url('../../Icons/noun_hover.webp');
}

.back-button {
  background-image: url('../../Icons/back-icon.png');
  width: 40px;
  border: none;
  background-color: white;
  border-radius: 50%;
  background-size: 65%;
  background-repeat: no-repeat;
  background-position: 45%;
  cursor: pointer;
  transition-duration: 00.2s;
}

.back-button:hover {
  background-color: lightgray;
}

span {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 15px;
}

#posted-by {
  font-size: 14px;
  font-style: italic;
  margin-left: auto;
  margin-right: 10px;
}

.comments-container {
  width: 98%;
  margin-top: 10px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  border: 3px solid red;
  background-color: white;
}

.comment-item {
  max-width: 100%;
  width: 96%;
  padding-top: 20px;
  padding-bottom: 30px;
  padding-left: 2%;
  padding-right: 2%;
  border: 1px solid black;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
}

.username {
  margin-bottom: 5px;
  font-weight: 700;
  cursor: pointer;
}

.first-reply-layer {
  width: 93%;
  border-radius: 10px;
  margin-top: 20px;
  margin-bottom: 10px;
  padding-top: 15px;
  padding-bottom: 20px;
  padding-left: 4%;
  padding-right: 2%;
  background-color: rgba(211, 211, 211, 0.315);
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
}

.second-reply-layer {
  width: 93%;
  border-radius: 10px;
  margin-top: 20px;
  margin-bottom: 10px;
  padding-top: 15px;
  padding-bottom: 20px;
  padding-left: 4%;
  padding-right: 2%;
  background-color: rgba(211, 211, 211, 0.884);
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
}

/* Smartphones/Tablets (portrait) ----------- */
@media only screen
and (max-width : 768px) {
  nav {
    display: none;
  }

  main {
    width: 100%;
    padding-top: 60px;
  }

  .main-header {
    width: 95%;
    height: 60px;
  }

  .main-header h2 {
    display: none;
  }

  .main-header #brand {
    display: none;
  }

  .main-header input {
    display: inline-block;
    width: 50%;
    height: 19px;
    margin-left: 4%;
    margin-top: 15px;
  }

  .main-header select {
    display: inline-block;
    width: 35%;
    height: 25px;
    margin-left: 4%;
    margin-top: 15px;
  }

  .reddit-post {
    height: 400px;
  }

  .sub {
    height: 25px;
  }

  .content {
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-around;
  }

  .content-title {
    width: 100%;
    text-align: left;
  }

  .content-media {
    width: 100%;
    height: 85%;
    text-align: left;
  }

  .content-link {
    max-width: fit-content;
  }

  .options {
    height: 25px;
  }
}