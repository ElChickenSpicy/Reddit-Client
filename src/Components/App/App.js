import React from 'react';
import './App.css';
import { Navbar } from '../Navbar/Navbar';
import { Main } from '../Main/Main';

export class App extends React.Component {
  render() {
    return (
      <div className="top-container">
        <Navbar />
        <Main />
      </div>
    )
  }
};

