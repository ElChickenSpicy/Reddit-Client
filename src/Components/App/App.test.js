import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

//Setup
let app = shallow(<App />);

//Test
describe('APP component as a whole', () => {
  it('render the App Component correctly', () => {   
    expect(app)
    .toMatchSnapshot();
  });
});
