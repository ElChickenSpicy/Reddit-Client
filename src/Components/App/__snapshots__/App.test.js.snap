// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP component as a whole render the App Component correctly 1`] = `
<div
  className="top-container"
>
  <Navbar />
  <Main />
</div>
`;
